File: src/com/main/tasks/gcd_lcm/Gcd.kt
package com.main.tasks.gcd_lcm

import com.main.tasks.measure
import java.io.File

fun main() {
    val (a, b) = File("numbers.txt").readLines().map { it.toInt() }
    measure { print(findGcd(a, b)) }
}

fun findGcd(a: Int, b: Int): Int {
    if (a == 0) return b
    if (b == 0) return a
    return findGcd(b, a % b)
}

File: src/com/main/tasks/gcd_lcm/dart/gcd.dart
import 'dart:io';

Future<void> main() async {
  final lines = await File('../../../../../../numbers.txt').readAsLines();
  final a = lines.map((e) => int.parse(e)).first;
  final b = lines.map((e) => int.parse(e)).last;
  print(findGcd(a, b));
}

int findGcd(int a, int b) {
  if (a == 0) return b;
  if (b == 0) return a;
  return findGcd(b, a % b);
}
